docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,542 @@
+Volume path names are examples only and not necessarily as specified.
+
+| Path | Resource |
+| -------------------- | -------------------|
+| cpu_limit | resources.limits.cpu |
+| memory_limit | resources.limits.memory|
+| cpu_request | resources.requests.cpu |
+| memory_request | resources.requests.memory |
+
+Since environment variables are container scoped, the container name must
+be specified.
+
+#### Examples
+These examples show how to use no selectors approach with environment variables and volume plugin.
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dapi-test-pod
+spec:
+  containers:
+    - name: test-container
+      image: gcr.io/google_containers/busybox
+      command: [ "/bin/sh","-c", "env" ]
+      resources:
+        requests:
+          memory: "64Mi"
+          cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      env:
+        - name: CPU_LIMIT
+          valueFrom:
+            resourceFieldRef:
+              resource: resources.limits.cpu
+```
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kubernetes-downwardapi-volume-example
+spec:
+  containers:
+    - name: client-container
+      image: gcr.io/google_containers/busybox
+      command: ["sh", "-c","while true; do if [[ -e /etc/labels ]]; then cat /etc/labels; fi; if [[ -e /etc/annotations ]]; then cat /etc/annotations; fi; sleep 5; done"]
+      resources:
+        requests:
+          memory: "64Mi"
+          cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      volumeMounts:
+        - name: podinfo
+          mountPath: /etc
+          readOnly: false
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "cpu_limit"
+            resourceFieldRef:
+              containerName: client-container
+              resource: resources.limits.cpu
+```
+
+#### Validations
+
+For APIs with no selectors, verify that the resource strings are valid and is one
+of `resources.limits.cpu`, `resources.limits.memory`, `resources.requests.cpu` and `resources.requests.memory`.
+Also verify that container name is provided with volumes.
+
+## Pod and container level resource access
+Pod level resources (like `metadata.name`, `status.podIP`) will always be accessed with `type ObjectFieldSelector` object in
+all approaches. Container level resources will be accessed by `type ObjectFieldSelector`
+with full selector approach; and by `type ContainerFieldRef` and `type ResourceFieldRef`
+with partial and no selectors approaches, respectively. The following table
+summarizes resource access with these approaches.
+
+| Approach | Pod resources| Container resources |
+| -------------------- | -------------------|-------------------|
+| Full selectors | `ObjectFieldSelector` | `ObjectFieldSelector`|
+| Partial selectors | `ObjectFieldSelector`| `ContainerFieldRef` |
+| No selectors | `ObjectFieldSelector`| `ResourceFieldRef` |
```

Comment:
I think I can change it to magic keys approach instead of saying no selectors if that sounds better.